components/director/internal/runtimemapping/token_verifier.go

```diff
@@ -98,10 +98,14 @@ func (f *jwksFetch) GetKey(ctx context.Context, token *jwt.Token) (interface{},
 		return nil, errors.Wrap(err, "while getting the key ID")
 	}
 
-	keys := jwksSet.LookupKeyID(keyID)
-	for _, key := range keys {
-		if key.Algorithm() == token.Method.Alg() {
-			return key.Materialize()
+	if keys, isFound := jwksSet.LookupKeyID(keyID); isFound {
+		if keys.Algorithm() == token.Method.Alg() {
```

Comment:
this if can be added to the if above like `isFound && keys.Algorithm() == token.Method.Alg()`